src/components/svgs/nodes/szmetro-num-line-badge.tsx

```diff
@@ -0,0 +1,165 @@
+                </text>
+                <text className="rmp-name__en" x={enX} y="13.5" fontSize="3" fill={fgColor}>
+                    {isBranch ? 'Branch' : ''} Line {num}
+                </text>
+                {/* Below is an overlay element that has all event hooks but can not be seen. */}
+                <rect
+                    fill="white"
+                    fillOpacity="0"
+                    x="0"
+                    width={NUM_WIDTH + 23}
+                    height="16"
+                    rx="2"
+                    onPointerDown={onPointerDown}
+                    onPointerMove={onPointerMove}
+                    onPointerUp={onPointerUp}
+                    style={{ cursor: 'move' }}
+                />
+            </g>
+        ),
+        [id, x, y, num, ...color, isBranch, onPointerDown, onPointerMove, onPointerUp]
+    );
+};
+
+/**
+ * BjsubwayNumLineBadge specific props.
+ */
+export interface SzmetroNumLineBadgeAttributes extends AttributesWithColor {
+    num: number;
+    isBranch: boolean;
+}
+
+const defaultSzmetroNumLineBadgeAttributes: SzmetroNumLineBadgeAttributes = {
+    num: 1,
+    color: [CityCode.Beijing, 'bj1', '#c23a30', MonoColour.white],
+    isBranch: false,
+};
+
+const szmetroNumLineBadgeFields = [
+    {
+        type: 'input',
+        label: 'panel.details.nodes.common.num',
+        value: (attrs?: SzmetroNumLineBadgeAttributes) => (attrs ?? defaultSzmetroNumLineBadgeAttributes).num,
+        validator: (val: string) => !Number.isNaN(val),
+        onChange: (val: string | number, attrs_: SzmetroNumLineBadgeAttributes | undefined) => {
+            // set default value if switched from another type
+            const attrs = attrs_ ?? defaultSzmetroNumLineBadgeAttributes;
+            // return if invalid
+            if (Number.isNaN(val)) return attrs;
+            // set value
+            attrs.num = Number(val);
+            // return modified attrs
+            return attrs;
+        },
+    },
+    {
+        type: 'custom',
+        label: 'color',
+        component: (
+            <ColorField
+                type={MiscNodeType.SzmetroNumLineBadge}
+                defaultTheme={defaultSzmetroNumLineBadgeAttributes.color}
+            />
+        ),
+    },
+    {
+        type: 'switch',
+        label: 'panel.details.stations.bjsubwayBasic.open',
+        oneLine: true,
+        isChecked: (attrs?: SzmetroNumLineBadgeAttributes) => (attrs ?? defaultSzmetroNumLineBadgeAttributes).isBranch,
+        onChange: (val: boolean, attrs_: SzmetroNumLineBadgeAttributes | undefined) => {
+            // set default value if switched from another type
+            const attrs = attrs_ ?? defaultSzmetroNumLineBadgeAttributes;
+            // set value
+            attrs.isBranch = val;
+            // return modified attrs
+            return attrs;
+        },
+    },
+];
+
+const attrsComponent = () => (
+    <RmgFieldsFieldSpecificAttributes
```

Comment:
`RmgFieldsFieldSpecificAttributes` is deprecated. Use standard React components with `RmgFields` :)

Reply:
More references could be found at https://github.com/railmapgen/rmp/blob/5ae7d62be08ac96a220f6c3492bbc73178e8d683/src/components/svgs/stations/shmetro-basic.tsx#L118